coordinator/src/services/translation/signing.ts

```diff
@@ -0,0 +1,147 @@
+    }
+  } as XmlJs.ElementCompact)
+}
+
+export function convertFragmentsToDisplayableFormat(fragments: Fragments): Display {
+  const prescriptionStartDate = fragments.time._attributes.value
+
+  const patient = fragments.recordTarget.Patient
+  const patientPerson = patient.patientPerson
+
+  const patientName =
+    joinCoreText(" ",
+      patientPerson.name[0].prefix,
+      patientPerson.name[0].given,
+      patientPerson.name[0].family
+    )
+  const patientDob = patientPerson.birthTime._attributes.value
+  const patientGender = getGenderFromCode(patientPerson.administrativeGenderCode)
+  const patientNhsNumber =  patient.id._attributes.extension
+  const patientAddress = joinCoreText(", ",
+    patient.addr[0].streetAddressLine,
+    patient.addr[0].postalCode
+  )
+
+  const prescriber = fragments.agentPerson
+  const prescriberPerson = prescriber.agentPerson
+  const prescribingOrg = prescriber.representedOrganization
+
+  const prescriberName = joinCoreText(" ",
+    prescriberPerson.name.prefix,
+    prescriberPerson.name.given,
+    prescriberPerson.name.family
+  )
+  const prescriberCode = prescriber.id._attributes.extension
+  const prescriberOrg = `${prescribingOrg.name._text} (${prescribingOrg.id._attributes.extension})`
+  const prescriberAddress = joinCoreText(", ",
+    prescribingOrg.addr.streetAddressLine,
+    prescribingOrg.addr.postalCode
+  )
+
+  const medication = fragments.pertinentLineItem
+    .map((medication) => {
+      return new DisplayMedication({
+        name: medication.product.manufacturedProduct.manufacturedRequestedMaterial.code._attributes.displayName,
+        quantity: medication.component.lineItemQuantity.quantity._attributes.value,
+        dose: medication.pertinentInformation2.pertinentDosageInstructions.value
+      })
+    })
+
+  const display: Display = {
+    prescriptionStartDate: prescriptionStartDate,
+    patientName: patientName,
+    patientDob: patientDob,
+    patientGender: patientGender,
+    patientNhsNumber: patientNhsNumber,
+    patientAddress: patientAddress,
+    prescriberName: prescriberName,
+    prescriberCode: prescriberCode,
+    prescriberOrg: prescriberOrg,
+    prescriberAddress: prescriberAddress,
+    medication: medication
+  }
+
+  return display
+}
+
+/* eslint-disable  @typescript-eslint/no-explicit-any */
+function joinCoreText(seperator: string, ...values: any[]) : string {
+  return values.map(unpackCoreText).filter(Boolean).join(seperator)
+}
+
+function unpackCoreText(value: core.Text | core.Text[]) : string {
+  if (isCoreTextArray(value)) {
+    return value.map(v => v._text).join(" ")
+  } else {
+    const text = value as core.Text
+    return text !== undefined ? text._text : ""
+  }
+}
+
+function isCoreTextArray(value: core.Text | core.Text[]) : value is core.Text[] {
+  const valueArray = value as core.Text[]
+  return valueArray !== undefined && valueArray.length > 0
+}
+
+function getGenderFromCode(code: codes.SexCode): string {
```

Comment:
already have this function defined in demographics.ts on line 90 - maybe a good idea to collect these into the common.ts file?

Reply:
this is actually the reverse of this function back from hl7v3 to fhir. It's very specific to the signing use-case so it made more sense to leave it here and not to pollute fhir to hl7v3 mappings with the reverse function when it's not used as all the other mappings are

Reply:
ahh of course - my bad